### ISSUE TITLE
Supporting SCAPE systems

### ISSUE THREAD

**Issue by henrypinkard** [micro-manager/LightSheetManager#37]
Since the UI/Acquisition engine/etc are now working with diSPIM, what is needed for SCAPE should now be mostly in place. But first we will need to duplicate/refactor the non-diSPIM-specific stuff.

SCAPE will need its own function in the LSM acquisition code. Right now the diSPIM one is almost entirely full of stuff that is shared between the two modalities:

https://github.com/micro-manager/LightSheetManager/blob/07938f1504353699c94e689d396048eb5267f579/src/main/java/org/micromanager/lightsheetmanager/model/AcquisitionEngine.java#L369

In order to refactor it, the [AcquisitionSettings for SCAPE](https://github.com/micro-manager/LightSheetManager/blob/main/src/main/java/org/micromanager/lightsheetmanager/api/internal/DefaultAcquisitionSettingsSCAPE.java) will need to be filled out:

It seems that the [analogous class for diSPIM ](https://github.com/micro-manager/LightSheetManager/blob/main/src/main/java/org/micromanager/lightsheetmanager/api/internal/DefaultAcquisitionSettingsDISPIM.java) is mostly full of settings that are common to the two modalities. So pulling out the common settings would be a first step. What do you think @bls337?

@fadero How do you currently send instructions to hardware to kick off a scan?

**Comment by nicost** [comment]
We currently use the Utility DAZ device to make the  DAC controlling the galvo mirror appear as a Z drive that can be sequenced.  Tanner sets up a ZStack in the MDA window (with multiple channels, and/or time points if so desired), and that outputs the desired data.  Light sheet plugin not really needed.

**Comment by bls337** [comment]
I will figure out what settings are necessary for SCAPE.

There is an empty runAcquisitionSCAPE method to put the acquisition code into, the requestRun method switches based on the microscope geometry type.

**Comment by bls337** [comment]
@henrypinkard

Right now LSMAcquisitionEvents is hard-coded to use DefaultAcquisitionSettingsDISPIM objects, we could make it work with SCAPE settings by using the interface and downcasting the type inside the method.

Should we use LSMAcquisitionEvents to generate the acquisition events for all microscope geometry types?

It seems like most of the acquisition settings are generic enough... do you anticipate that any microscope geometry type will have a vastly different setup in LSMAcquisitionEvents?

**Comment by henrypinkard** [comment]
Yes, I think LSMAcquisitionEvents can be reused for all. The only difference I see is the number of cameras, but this can be set through the function arguments so I don't think anything would need to change.


> Right now LSMAcquisitionEvents is hard-coded to use DefaultAcquisitionSettingsDISPIM objects, we could make it work with SCAPE settings by using the interface and downcasting the type inside the method.

I think that makes sense

Though maybe this is less of a priority than I thought If Nico doesn't need to use it (depending on Brian's needs)

**Comment by bls337** [comment]
A lot of work has been done to support SCAPE, to the point that I will need to revisit diSPIM and fix things. I've been trying to keep diSPIM somewhat current with the SCAPE side, but it has fallen behind. The current priority is SCAPE. The current name of the `LSMAcquisitionEvents` class is `LightSheetEventAdapter` and it uses `ScapeAcquisitionSettings` for now.

**Comment by bls337** [comment]
This work ties into the API improvement plan, moving some of the builders to the base `AcquisitionSettings` class will let `LightSheetEventAdapter` work generically across devices by taking `AcquisitionSettings` as a parameter instead of `ScapeAcquisitionSettings`.

A bonus is that new `AcquisitionSettings` types gain a lot of basic functionality for free.

**Comment by bls337** [comment]
#454 makes  the currently used methods on `LightSheetEventAdapter` geometry agnostic, it turns out there is one commented method that would need to be rewritten to to support diSPIM, but that is for another issue. I think this more or less concludes this issue, SCAPE is supported, diSPIM will need a revisit when I have a system built up in my office to test.
